### PR TITLE
ADOP-211 change

### PR DESCRIPTION
**Pull request by pankajhmcts**
### Change description
ADOP-211 change 

### JIRA link
https://tools.hmcts.net/jira/browse/ADOP-211
**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x ] No

### CODE REVIEW

File: src/main/steps/children/social-worker/content.test.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+jest.mock('../../../app/form/validation');
+
+const CY = 'cy';
+const EN = 'en';
+
+const enContent = {
+  section: 'Your adoption agency or local authority details',
+  title: "Details about the child's social worker",
+  line1: 'You can get these details from your adoption agency or local authority.',
+  socialWorkerNameLabel: "Social worker's name",
```
Comment (abidfs) [inline src/main/steps/children/social-worker/content.test.ts:18]: we are not specifying Label explicitly
could you please make it simple
```suggestion
  socialWorkerName: "Social worker's name",
```

This applies to below attributes as well

File: src/main/steps/children/social-worker/content.test.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+};
+
+const cyContent = {
+  section: 'Your adoption agency or local authority details (in Welsh)',
+  title: "Details about the child's social worker (in Welsh)",
+  line1: 'You can get these details from your adoption agency or local authority. (in Welsh)',
+  socialWorkerNameLabel: "Social worker's name (in Welsh)",
+  socialWorkerPhoneNumberLabel: "Social worker's phone number (in Welsh)",
+  socialWorkerEmailLabel: "Social worker's email address (in Welsh)",
+  socialWorkerTeamEmailLabel: "Social worker's team email address (if known) (in Welsh)",
```
Comment (abidfs) [inline src/main/steps/children/social-worker/content.test.ts:44]: same as english

File: src/main/steps/children/social-worker/content.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import { isEmailValid, isFieldFilledIn, isPhoneNoValid } from '../../../app/form/validation';
+
+export const en = (): Record<string, unknown> => ({
+  section: 'Your adoption agency or local authority details',
+  title: "Details about the child's social worker",
+  line1: 'You can get these details from your adoption agency or local authority.',
+  socialWorkerNameLabel: "Social worker's name",
+  socialWorkerPhoneNumberLabel: "Social worker's phone number",
+  socialWorkerEmailLabel: "Social worker's email address",
+  socialWorkerTeamEmailLabel: "Social worker's team email address (if known)",
```
Comment (abidfs) [inline src/main/steps/children/social-worker/content.ts:12]: could you please review label from name

File: src/main/steps/children/social-worker/content.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      classes: 'govuk-label govuk-!-width-two-thirds',
+      label: l => l.socialWorkerEmailLabel,
+      labelSize: null,
+      validator: value => isFieldFilledIn(value) || isEmailValid(value),
+    },
+    socialWorkerTeamEmail: {
+      type: 'text',
+      classes: 'govuk-label govuk-!-width-two-thirds',
+      label: l => l.socialWorkerTeamEmailLabel,
+      labelSize: null,
```
Comment (abidfs) [inline src/main/steps/children/social-worker/content.ts:81]: I think we would need a validation here, if they are `socialWorkerTeamEmail` then it has to be valid email

**Review by abidfs — CHANGES_REQUESTED** [review]
Looks good, however few tiny bits before we merge this

File: src/main/steps/children/social-worker/content.ts (hunk 4 above)
Comment (abidfs) [reply src/main/steps/children/social-worker/content.ts:81]:
```
validator: value => value ? isEmailValid(value) : undefined,
```
This should work

also remember to add
```
 socialWorkerTeamEmail:{
      invalid: 'Enter an email address in the correct format, like name@example.com (in Welsh)',
    }
```
in enContent and cyContent

File: src/main/steps/children/social-worker/content.test.ts [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    const generatedContent = generateContent(commonContent);
+    const form = generatedContent.form as FormContent;
+    const fields = form.fields as FormFields;
+    const socialWorkerTeamEmail = fields.socialWorkerTeamEmail;
+
+    expect(socialWorkerTeamEmail.type).toBe('text');
+    expect((socialWorkerTeamEmail.label as Function)(generateContent(commonContent))).toBe(
+      enContent.socialWorkerTeamEmail
+    );
+  });
```
Comment (abidfs) [inline src/main/steps/children/social-worker/content.test.ts:151]: unit tests missing for validator

File: src/main/steps/children/social-worker/content.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      label: l => l.socialWorkerEmail,
+      labelSize: null,
+      validator: value => isFieldFilledIn(value) || isEmailValid(value),
+    },
+    socialWorkerTeamEmail: {
+      type: 'text',
+      classes: 'govuk-label govuk-!-width-two-thirds',
+      label: l => l.socialWorkerTeamEmail,
+      labelSize: null,
+      validator: value => isEmailValid(value),
```
Comment (abidfs) [inline src/main/steps/children/social-worker/content.ts:88]: This is not working as expected, can you make it as per the above comment

File: src/main/steps/children/social-worker/content.test.ts [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    const socialWorkerTeamEmail = fields.socialWorkerTeamEmail;
+
+    expect(socialWorkerTeamEmail.type).toBe('text');
+    expect((socialWorkerTeamEmail.label as Function)(generateContent(commonContent))).toBe(
+      enContent.socialWorkerTeamEmail
+    );
+
+    const errors = generatedContent.errors as any;
+    expect(errors.socialWorkerTeamEmail.invalid).toEqual(
+      'Enter an email address in the correct format, like name@example.com'
```
Comment (abidfs) [inline src/main/steps/children/social-worker/content.test.ts:163]: content testing should not be here

Comment (abidfs) [reply src/main/steps/children/social-worker/content.test.ts:163]: Please run test:coverage and see the report

Comment (pankajhmcts) [reply src/main/steps/children/social-worker/content.test.ts]: --------------------------------------------------------|---------|----------|---------|---------|-----------------------------------------------------
File                                                    | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                                   
--------------------------------------------------------|---------|----------|---------|---------|-----------------------------------------------------
All files                                               |   96.02 |    85.21 |    92.7 |   95.73 |
 main/steps/children/social-worker                      |     100 |    83.33 |     100 |     100 |                                                     
  content.ts                                            |     100 |    83.33 |     100 |     100 | 88

**Review by pankajhmcts — COMMENTED** [review]
all review comments taken care.

**Review by abidfs — APPROVED** [review]

**Pull request merged**